### PR TITLE
Dynamic regularization

### PR DESCRIPTION
**Pull request by mlwahl**

### CODE REVIEW

File: de/model.jl [hunk 1]
```diff
@@ -127,12 +164,12 @@ function costG!(G, pIn, exp_data)
...
     # Regularization
-    @. G[1:6889] += 10 * (0.01 * sign(pIn[1:6889]))
+    @. G[1:6889] += 1000 * (0.01 * sign(pIn[1:6889]))
     w = reshapeParams(pIn)[1]
     T₀ = w' * w - I
-    temp = 10 * vec(2 / norm(T₀) * w * T₀)
+    temp = 10000 * vec(2 / norm(T₀) * w * T₀)
     @. G[1:6889] += temp
-
+    G += Zygote.gradient(costEigvals, pIn)[1]
```
Comment (aarmey) [inline de/model.jl:172]: I bet it's coming from the gradient calculation. Try surrounding the Zygote call here with `real.()` to only keep the real component.

Comment (mlwahl) [reply de/model.jl:172]: Same error.. I also managed to get the same error outside of `runOptim` just calling Zygote.gradient on its own

Comment (aarmey) [reply de/model.jl:172]: Try after the change I just made.

Comment (mlwahl) [reply de/model.jl:172]: Seems to have worked! I can work on resolving merge conflicts

File: de/model.jl [hunk 2]
```diff
@@ -49,14 +50,46 @@ end
...
+
+    return Diagonal(ε .* (sech.(w * u) .^ 2)) * w .- Diagonal(ɑ)
+end
+
+" Regularization: Calculate cost as sum of all but largest complex components of eigenvalues. "
+function costEigvals(pIn)
+    u = solveODE(pIn)
+    jacobian = ODEjac(u, pIn, 10000)
+    im_comps = abs.(imag(eigen(jacobian).values))
+    return norm(sum(im_comps) - maximum(im_comps))
```
Comment (aarmey) [inline de/model.jl:72]: Don't think you should need the norm, since you already took the sum.

**Pull request merged**